pkg/kubelet/dockertools/images.go

```diff
@@ -18,54 +18,85 @@ package dockertools
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/golang/glog"
 
 	dockertypes "github.com/docker/engine-api/types"
 	runtime "k8s.io/kubernetes/pkg/kubelet/container"
-	"k8s.io/kubernetes/pkg/util/sets"
 )
 
 // imageStatsProvider exposes stats about all images currently available.
 type imageStatsProvider struct {
+	sync.Mutex
```

Comment:
Discussed with @timstclair, and he said it is possible that summary api is called by multiple threads, so I added the lock here. :)